inline-c/src/Language/C/Types/Parse.hs

```diff
@@ -349,12 +359,38 @@ cidentifier = token cidentifier_no_lex
 
 type_name :: CParser i m => m CIdentifier
 type_name = try $ do
-  s <- ident cIdentStyle <?> "type name"
+  ctx <- ask
+  s <- ident' (cpcEnableCpp ctx) cIdentStyle <?> "type name"
   typeNames <- asks cpcTypeNames
   unless (HashSet.member s typeNames) $
     unexpected $ "identifier  " ++ unCIdentifier s
   return s
 
+templateParser :: (Monad m, CharParsing m, CParser i m) => IdentifierStyle m -> m TypeSpecifier
+templateParser s = parse'
+  where
+    parse' = do
+      id' <- cidentParserWithNamespace
```

Comment:
Why do we need this rather than just using `cpcParseIdent`? That should include namespaces now, right?

Reply:
It should include namespaces to parse the format like "std::vector<int>".